src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -68,6 +73,12 @@ internal void SetErrors(IEnumerable<IDocumentError> errors)
         /// </summary>
         public ReadOnlySymbolTable Symbols { get; set; }
 
+        /// <summary>
+        /// Parameters are the subset of symbols that must be passed in Eval() for each evaluation. 
+        /// This lets us associated the type in Check()  with the values in Eval().
+        /// </summary>
+        internal ReadOnlySymbolTable Parameters { get; set; }

```

Comment:
>Parameters [](http://example.com/codeflow?start=37&length=10)

Why do we need a distinct object?
Couldn't we store parameters in Symbols (with a flag/option to declare them as parameters)

Reply:
We could have done that as an alternative design, but I don't think it would have been any simpler.